### ISSUE TITLE
Integración TVH

### ISSUE THREAD

**Issue by rioreal2** [MovistarTV/tv_grab_es_movistartv#8]
Buenos días

Estoy tratando de integrar el grabber en el TVHeadEnd que tengo instalado en un NAS Synology pero he probado con symlink, metiendo directamente el fichero en /usr/bin (tanto con extensión como sin ella), he revisado permisos, etc. pero TVH no lo encuentra.

Sabes que puede estar mal?

Gracias

**Comment by rioreal2** [comment]
Gracias por las respuestas.

Tengo Python 2.7 instalado y XMLTV también, de hecho tengo otros grabbers funcionando correctamente aunque ninguno de ellos para Python.

He comprobado todo lo que me dices pero sigue sin aparecer.

Me estoy volviendo loco porque después de mucho buscar, tu solución me encanta pero no soy capaz de ponerla en marcha.

He probado a ejecutar el grabber y generar un m3u y funciona bien pero al intentarlo con un xml (tv_grab_es_movistartv.py --output /guia.xml) solo me ha funcionado 1 vez. El resto de intentos, tarda mucho y crea más de 20.000 ficheros .json en /cache/programs.

Un saludo

**Comment by MovistarTV** [comment]
Sí, puede tardar horas. En esos JSON está la información extendida de los programas. El servicio web que ha montado Movistar para IPTV es una vergüenza...

Casi seguro que tienes un problema de permisos. El usuario que ejecuta TVHeadend no tiene permiso para ejecutar el grabber... ¿El usuario es "hts"? Si no es así, cambia esta orden (hts por el nombre correcto) y lánzala:

`sudo -u hts tv_find_grabbers --verbose`

Compara la salida con:

`sudo tv_find_grabbers --verbose`

Y pégalo todo aquí.

**Comment by rioreal2** [comment]
No tengo la herramienta tv_find_grabbers... He tratado de instalarla pero no consigo que funciones en my NAS. Por lo que he podido investigar, para compilar XMLTV para mi sistema necesito PERL 5.6 pero no existe esa version en Synology, el PERL que tengo instalado es 5.2

**Comment by MovistarTV** [comment]
No te preocupes, acabo de echar un ojo a las fuentes de TVHeadend y en caso de no encontrar `tv_find_grabbers` usa una búsqueda interna (función [`_xmltv_load_grabbers`](https://github.com/tvheadend/tvheadend/blob/master/src/epggrab/module/xmltv.c))

Por lo que veo podría fallar:

- Que `/usr/bin` no está en el PATH: lo sabes con `echo $PATH`.
- Se utiliza la función `stat` de C y ésta requiere que todos los directorios en la ruta al archivo tengan permiso de ejecución. El grabber está en /usr/bin, tanto `/usr` como `/usr/bin` deberían tener el permiso `x` para todos los usuarios. Compruébalo haciendo un `ls -l` por ejemplo.
- Que el propio grabber no sea ejecutable para el usuario que corre TVHeadend, que es lo más probable.

No se... además de los permisos intenta cambiar también el propietario. Lo ideal es que fuese el usuario de TVHeadend (no sé cual es en Synology) pero sino:

`chown root:root /usr/bin/tv_grab_es_movistartv.py`
`chmod 777 /usr/bin/tv_grab_es_movistartv.py`

Y a ver qué pasa...

Un saludo.

**Comment by rioreal2** [comment]
Por fin!! Como bien sospechabas, era un tema de propietario/grupo. Al parecer el usuario de TVH en Synology pertenece al grupo "sc-media". Asi que tema resuelto.

Abusando de tu conocimiento, tengo la duda si en TVH tengo que dar de alta una red tomando los canales de un m3u y mapear los servicios como hacia con otros métodos o tu script crea la red y los muxes/servicios/canales.

Mil gracias

**Comment by MovistarTV** [comment]
:+1: 

Si has usado el asistente que aparece después de la instalación o lo vas a usar, en principio no hace falta que la crees porque es justo lo que hace el asistente, aunque sí deberías editarla y activar la opción `Accept zero value for TSID` como mínimo...

Mi red está así (una vez que TVHeadend ha encontrado todos los muxes, echa un ojo más abajo):

![network](https://user-images.githubusercontent.com/29414832/44935159-e339ef00-ad6f-11e8-8c35-703561c1d598.png)

Te recomiendo que eches un vistazo [a este post](https://www.adslzone.net/foro/movistar-tv-imagenio.38/obtener-lista-canales-programacion-movistar-tv.359916/page-10#post-3033880). Enlaza a una guía que explica cómo configurar la red desde cero.

Un saludo.

**Comment by skramouche** [comment]
Se que iuse esta cerrado, pero me estoy volviendo loco con este mismo problema y no doy con ello, he probado a cambiar permisos, propietario todo y no consigo que se vea por ningun lado.

**Comment by skramouche** [comment]
Bueno, parece que consegui resolver que no me apareciera en synology, dejare la manera por si a alguien mas le pasa, que no se vuelva loco. Resulta que ni permisos ni nada, que no aparecia, le deje ejecutable(755) y propietario y gupo root. Fui al archivo config del epggrabber (/volume1/@appstore/tvheadend/var/epggrab/config) y meti el graber a mano:

> 		"/usr/local/bin/tv_grab_es_movistartv.py": {
			"class": "epggrab_mod_int_xmltv",
			"dn_chnum": 0,
			"args": "--tvheadend /volume1/GuiaTv/MovistarTV.m3u",
			"name": "Spain (MovistarTV)",
			"type": "Internal",
			"enabled": true,
			"priority": 3
		},<
Y ahora funciona perfectamente, gracias y espero que a alguien le sirva.

**Comment by vsc55** [comment]
Hola, buenas.
Yo también me volví un poco loco ya que en Raspbian tampoco me aparecía.
Aquí dejo la mi solución por si ha alguien más le pasa.
Fue tan sencillo como ejecutar `tv_find_grabbers` y reiniciar el servicio. :D

**Comment by vgallego65** [comment]
Me estoy volviendo loco y no consigo que me salga el modulo grabador en la configuración de tvheadend. La verdad es que estoy muy verde, he tratado de leer todo lo posible pero no llego a nada.
Tengo otro graber funcionando y la ruta es /usr/local/bin/tv_grab_movistar-spain, por lo que he probado a mover también a ese directorio, ya que moviéndolo a /usr/bin y reiniciado no me aparece, pero tampoco he tenido éxito.
He aplicado algún comando como se explica mas arriba y el resultado ha sido este:
login as: admin
admin@192.168.1.100's password:
admin@synology:~$ echo $PATH
/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin
admin@synology:~$ echo $PATH
/sbin:/bin:/usr/sbin:/usr/bin:/usr/syno/sbin:/usr/syno/bin:/usr/local/sbin:/usr/local/bin
admin@synology:~$ ls -l
total 0
drwxrwxrwx+ 1 admin users 12 Apr  4  2016 CloudStation
drwxrwxrwx+ 1 admin users 12 Feb 13  2018 Drive
admin@synology:~$ login as: admin
admin@synology:~$ chown root:root /usr/local/bin/tv_grab_es_movistartv.py
chown: cannot access ‘/usr/local/bin/tv_grab_es_movistartv.py’: No such file or directory
admin@synology:~$ chown root:root /usr/local/bin/tv_grab_es_movistartv
chown: changing ownership of ‘/usr/local/bin/tv_grab_es_movistartv’: Operation not permitted
admin@synology:~$ chmod 777 /usr/local/bin/tv_grab_es_movistartv
chmod: changing permissions of ‘/usr/local/bin/tv_grab_es_movistartv’: Operation not permitted
admin@synology:~$ tv_find_grabbers
-sh: tv_find_grabbers: command not found
admin@synology:~$ cd /volume2/@appstore/tvheadend/var/epggrab/config
-sh: cd: /volume2/@appstore/tvheadend/var/epggrab/config: Not a directory
admin@synology:~$ tv_find_grabbers
-sh: tv_find_grabbers: command not found

Ya no se que mas comprobar, por favor ayudadme necesito hacer funcionar esto.

**Comment by MovistarTV** [comment]
Intenta meterlo a mano como hizo @skramouche:

[https://github.com/MovistarTV/tv_grab_es_movistartv/issues/8#issuecomment-589810733](url)

**Comment by vsc55** [comment]
Me parece que el módulo de "EPG Grabber" no viene incluido en el paquete que se instalas desde el centro de paquetes de Synology.

**Comment by vgallego65** [comment]
> Intenta meterlo a mano como hizo @skramouche:
> 
> [https://github.com/MovistarTV/tv_grab_es_movistartv/issues/8#issuecomment-589810733](url)

Pues no me queda nada claro como se hace eso, ya dije que a nivel comando, consola, Linux y demás estoy muy verde, a mi en principio no me deja acceder al archivo config en /volume2/@appstore/tvheadend/var/epggrab/config (Yo tengo todas las app del sistema en el volume2), pero intento entrar por consola como admin y no me deja acceder.

**Comment by vgallego65** [comment]
> Me parece que el módulo de "EPG Grabber" no viene incluido en el paquete que se instalas desde el centro de paquetes de Synology.

Hola, ¿Y lo puedo instalar de alguna forma?

**Comment by vsc55** [comment]
> 
> 
> > Me parece que el módulo de "EPG Grabber" no viene incluido en el paquete que se instalas desde el centro de paquetes de Synology.
> 
> Hola, ¿Y lo puedo instalar de alguna forma?

Echa un ojo a este enlace hablan algo del tema.
https://tvheadend.org/boards/12/topics/11954
